src/lib/broadcast.jl

```diff
@@ -263,8 +263,8 @@ end
 
   end
 
-  @adjoint CUDA.CuArray{N,T}(xs::Array) where {N,T} =
-    CUDA.CuArray{N,T}(xs), Δ -> (convert(Array, Δ), )
+  @adjoint (::Type{<:CUDA.CuArray{T,N}})(xs::Array) where {T,N} =
+    CUDA.CuArray{T,N}(xs), Δ -> (convert(Array, Δ), )
```

Comment:
Shouldn't this apply the full type in the forward pass, not just the abstract type with two typevars?
```suggestion
  @adjoint (::Type{T})(xs::Array) where {T<:CUDA.CuArray} =
    T(xs), Δ -> (convert(Array, Δ), )
```
Also, this PR has no tests.

Reply:
That parameterized version gets called directly, so we need a constructor that also has these parameters.

Reply:
I don't think that's true. You are claiming that this would fail, because it similarly omits a type parameter:
```
julia> (::Type{T})(x::Char) where {T<:Array} = "called";

julia> Array{Int}('a')
"called"
```

Reply:
I didn't claim that. Either way, the rule you want is already there https://github.com/FluxML/Zygote.jl/blob/807e5e5b3e99e8f5bb60e721b77d9a95261e62a9/src/lib/broadcast.jl#L281